apps/internal/oauth/oauth.go

```diff
@@ -194,14 +195,28 @@ func (d DeviceCode) Token(ctx context.Context) (accesstokens.TokenResponse, erro
 	}
 }
 
-var waitRE = regexp.MustCompile("(authorization_pending|slow_down)")
+type DeviceCodeError struct {
+	Error string `json:"error"`
+}
 
-// TODO(msal): This is freaking terrible. The original just looked for the exact word in the error output.
-// I doubt this worked. I don't know if the service really does this, but it should send back a structured
-// error response. Anyways, I updated this to search the entire return error message, which will be the body
-// of the return.
 func isWaitDeviceCodeErr(err error) bool {
-	return waitRE.MatchString(err.Error())
+	var c errors.CallErr
+	if errors.As(err, &c) {
+		if c.Resp.StatusCode == 400 {
```

Comment:
Lot of nesting here.  While this is mostly stylistic, most Go programs flip the conditions to avoid the nesting (some people call this the "happy line").

Reply:
Removed nesting in this commit https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/178/commits/5fbd566d1d01afddf610abb63fa29e05fff314ce